### PR TITLE
factory contract integration testcases

### PR DESCRIPTION
**Pull request by Abhishek-1857**
# Pull Request

## Description
Added integration test cases for core functionalities in the factory contract, including `create pool`, `add liquidity`, `register token`, and `swap token`. These tests ensure correctness and robustness for essential contract operations, along with edge case validation for handling unexpected inputs or scenarios.

## Type of Change

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Documentation update
- [ ] Refactor/Chore

## Related Issue
None

## Testing
 integration tests added for factory contract

## Checklist

- [ ] Self-review completed
- [ ] Tests added/updated (if applicable)
- [ ] Documentation updated (if applicable)

## Screenshots (if applicable)
None

## Additional Notes
None

### CODE REVIEW

File: tests-integration/src/tests/factory.rs [hunk 1]
```diff
@@ -754,5 +758,1458 @@ fn test_add_liquidity() {
...
+        &factory,
+        pair_info,
+        BPS_100_PERCENT,
+        Some(30),
+        funds,
+    );
+}
+
+#[test]
+fn test_add_liquidity_with_invalid_timeout() {
```
Comment (joemonem) [inline tests-integration/src/tests/factory.rs]: This test is identical to the one above. It isn't providing an invalid timeout

File: tests-integration/src/tests/factory.rs [hunk 2]
```diff
@@ -754,5 +758,1458 @@ fn test_add_liquidity() {
...
+        faucet(
+            &chain,
+            chain.sender.as_str(),
+            token.amount.u128(),
+            token.token_type.clone(),
+            &mut funds,
+        );
+    }
+
+    add_liquidity(
```
Comment (joemonem) [inline tests-integration/src/tests/factory.rs:1542]: In cross-chain test cases we need to set assertions on the sending and receiving chain to make sure everything is working as expected. Errors on the receiving chain won't show here. 
This applies to all cases where were expect the transaction to work.

File: tests-integration/src/tests/factory.rs (hunk 1 above)
Comment (Abhishek-1857) [reply tests-integration/src/tests/factory.rs]: @joemonem I have fixed this testcase

File: tests-integration/src/tests/factory.rs (hunk 2 above)
Comment (Abhishek-1857) [reply tests-integration/src/tests/factory.rs:1542]: @joemonem  fixed some changes let me know if there should be any other assertions that need to be added

File: tests-integration/src/tests/factory.rs [hunk 3]
```diff
@@ -730,222 +730,362 @@ fn test_create_pool_with_funds() {
...
+                    token_2: TokenWithDenomAndAmount {
+                        token: Token::create("osmo".to_string()).unwrap(),
+                        amount: Uint128::from(100_000u128),
+                        token_type: euclid::token::TokenType::Native {
+                            denom: "osmo".to_string(),
+                        },
+                    },
+                },
+                slippage_tolerance_bps: 100, // 1% slippage tolerance
+                timeout: None,               // 10 minutes in seconds
```
Comment (joemonem) [inline tests-integration/src/tests/factory.rs:1157]: timeout is None, but the comment says 10 minutes in seconds

**Comment by joemonem** [comment]
Try merging the development branch into this one to resolve merge conflicts

File: tests-integration/src/tests/factory.rs (hunk 3 above)
Comment (Abhishek-1857) [reply tests-integration/src/tests/factory.rs:1157]: Fixed

**Review by joemonem — APPROVED** [review]
